vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -743,6 +859,61 @@ bool AdsImpl::TestSearchState(
   return is_search_engine;
 }
 
+void AdsImpl::TestCheckoutStateAndAdHistory(const std::string& url,
+    std::map<std::string, std::string> campaign_checkout_urls) {
+  // Iterate over checkout URLs
+  for (auto elem : campaign_checkout_urls) {
+    std::string campaign_id = elem.first;
+    std::string checkout_url = elem.second;
+
+    // If checkout URL matches and campaign found in ad history
+    // consider test successful
+    if (url == checkout_url) {
+      BLOG(INFO) << "Client in checkout state at " << url;
+      auto campaign_history = client_->GetCampaignHistory();
+
+      if (campaign_history.find(campaign_id) !=
+          campaign_history.end()) {
+        BLOG(INFO) << "Attribute conversion to campaign " << campaign_id;
+      }
+    }
+  }
+}
+
+void AdsImpl::ServeSegmentAdIfPurchaseIntent() {
+  // TODO(Moritz Haller): Have checks from AdsImpl::CheckAdReady
```

Comment:
We should not add business logic here to check if we are ready to serve an ad, ads should only be served from `AdsImpl::CheckReadyAdServe`